metal/cloud_test.go

```diff
@@ -84,17 +75,66 @@ func testGetValidCloud(t *testing.T) (*cloud, *store.Memory) {
 
 	// now just need to create a client
 	config := Config{
-		ProjectID: uuid.New().String(),
+		ProjectID:           uuid.New().String(),
+		LoadBalancerSetting: LoadBalancerSetting,
 	}
 	c, _ := newCloud(config, client)
 	ccb := &mockControllerClientBuilder{}
 	c.Initialize(ccb, nil)
-	validCloud = c.(*cloud)
-	return validCloud, backend
+
+	return c.(*cloud), backend
+}
+
+func TestLoadBalancerDefaultDisabled(t *testing.T) {
+	vc, _ := testGetValidCloud(t, "")
+	response, supported := vc.LoadBalancer()
+	var (
+		expectedSupported = false
+		expectedResponse  = response
+	)
+	if supported != expectedSupported {
+		t.Errorf("supported returned %v instead of expected %v", supported, expectedSupported)
+	}
+	if response != expectedResponse {
+		t.Errorf("value returned %v instead of expected %v", response, expectedResponse)
+	}
+}
+
+func TestLoadBalancerMetalLB(t *testing.T) {
+	t.Skip("Test needs a k8s client to work")
+	vc, _ := testGetValidCloud(t, "metallb:///metallb-system/config")
+	response, supported := vc.LoadBalancer()
+	var (
+		expectedSupported = true
+		expectedResponse  = response
+	)
+	if supported != expectedSupported {
+		t.Errorf("supported returned %v instead of expected %v", supported, expectedSupported)
+	}
+	if response != expectedResponse {
+		t.Errorf("value returned %v instead of expected %v", response, expectedResponse)
+	}
+}
+
+func TestLoadBalancerEmpty(t *testing.T) {
+	t.Skip("Test needs a k8s client to work")
+	vc, _ := testGetValidCloud(t, "empty://")
+	response, supported := vc.LoadBalancer()
+	var (
+		expectedSupported = true
+		expectedResponse  = response
+	)
+	if supported != expectedSupported {
+		t.Errorf("supported returned %v instead of expected %v", supported, expectedSupported)
+	}
+	if response != expectedResponse {
+		t.Errorf("value returned %v instead of expected %v", response, expectedResponse)
+	}
 }
```

Comment:
Skipping these because we don't have a k8s client plumbed through for the code paths to be able to run.